Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -5408,8 +5410,8 @@ void WebPageProxy::scalePageInViewCoordinates(double scale, const IntPoint& cent
         return;
 
     send(Messages::WebPage::DidScalePageInViewCoordinates(scale, centerInViewCoordinates));
-    forEachWebContentProcess([&] (auto& process, auto pageID) {
-        if (&process == &legacyMainFrameProcess())
+    forEachWebContentProcess([page = Ref { *this }, scale] (auto& process, auto pageID) {
```

Comment:
Seems we just need `NOESCAPE` on the lambda instead?

Reply:
Fixed! I also fixed some changes I made for `forEachWebContentProcess` in a previous patch as well.